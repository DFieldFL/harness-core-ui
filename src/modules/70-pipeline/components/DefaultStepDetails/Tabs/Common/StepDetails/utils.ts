export function encodeURIWithReservedChars(uri: string): string {
  return encodeURIComponent(uri).replace(/[;,/?:@&=+$#]/g, function (c) {
    return '%' + c.charCodeAt(0).toString(16)
  })
}
