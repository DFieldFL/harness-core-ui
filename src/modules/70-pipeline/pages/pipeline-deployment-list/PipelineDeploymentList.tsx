import React from 'react'
import { useParams } from 'react-router-dom'

import { useGetListOfExecutions } from 'services/cd-ng'
import { useStrings } from 'framework/exports'
import { Page } from '@common/exports'
import { useQueryParams } from '@common/hooks'
import type { PipelinePathProps } from '@common/interfaces/RouteInterfaces'

import ExecutionsFilter, { FilterQueryParams } from './ExecutionsFilter/ExecutionsFilter'
import ExecutionsList from './ExecutionsList/ExecutionsList'
import ExecutionsPagination from './ExecutionsPagination/ExecutionsPagination'
import css from './PipelineDeploymentList.module.scss'

const pollingIntervalInMilliseconds = 5_000

export interface PipelineDeploymentListProps {
  onRunPipeline(): void
}

export default function PipelineDeploymentList(props: PipelineDeploymentListProps): React.ReactElement {
  const { pipelineIdentifier, orgIdentifier, projectIdentifier, accountId } = useParams<PipelinePathProps>()
  const queryParams = useQueryParams<{ page?: string } & FilterQueryParams>()
  const page = parseInt(queryParams.page || '1', 10)
  const { getString } = useStrings()

  const { loading, data: pipelineExecutionSummary, error, refetch } = useGetListOfExecutions({
    queryParams: {
      accountIdentifier: accountId,
      projectIdentifier,
      orgIdentifier,
      pipelineIdentifiers: pipelineIdentifier
        ? [pipelineIdentifier]
        : queryParams.pipeline
        ? [queryParams.pipeline]
        : undefined,
      page: page - 1,
      searchTerm: queryParams.query,
      executionStatuses: queryParams.status ? [queryParams.status] : undefined
    },
    queryParamStringifyOptions: {
      arrayFormat: 'repeat'
    }
  })
  const hasFilters: boolean = !!queryParams.query || !!queryParams.pipeline || !!queryParams.status

  // Polling logic:
  //  - At any moment of time, only one polling is done
  //  - Only do polling on first page
  //  - When component is loading, wait until loading is done
  //  - When polling call (API) is being processed, wait until it's done then re-schedule
  React.useEffect(() => {
    const timeoutId = window.setTimeout(() => {
      if (page == 1 && !loading) {
        refetch()
      }
    }, pollingIntervalInMilliseconds)

    return () => {
      window.clearTimeout(timeoutId)
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [page, loading])

  return (
    <Page.Body
      className={css.main}
      loading={loading}
      error={error?.message}
      retryOnError={() => refetch?.()}
      noData={{
        when: () => !hasFilters && !pipelineExecutionSummary?.data?.content?.length,
        icon: 'cd-hover',
        message: getString('noDeploymentText'),
        buttonText: getString('runPipelineText'),
        onClick: props.onRunPipeline
      }}
    >
      <ExecutionsFilter onRunPipeline={props.onRunPipeline} />
      <ExecutionsList hasFilters={hasFilters} pipelineExecutionSummary={pipelineExecutionSummary} />
      <ExecutionsPagination pipelineExecutionSummary={pipelineExecutionSummary} />
    </Page.Body>
  )
}
