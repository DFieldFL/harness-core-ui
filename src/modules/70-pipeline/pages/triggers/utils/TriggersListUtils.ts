import type { IconName } from '@wings-software/uicore'
// temporary mock data
import type { AddDrawerMapInterface } from '@common/components/AddDrawer/AddDrawer'
import type { GetActionsListQueryParams } from 'services/pipeline-ng'
import type { StringKeys } from 'framework/strings'
import { TriggerTypes } from './TriggersWizardPageUtils'

export const GitSourceProviders: Record<
  GetActionsListQueryParams['sourceRepo'],
  { value: GetActionsListQueryParams['sourceRepo']; iconName: IconName }
> = {
  GITHUB: { value: 'GITHUB', iconName: 'github' },
  GITLAB: { value: 'GITLAB', iconName: 'service-gotlab' },
  BITBUCKET: { value: 'BITBUCKET', iconName: 'bitbucket' },
  AWS_CODECOMMIT: { value: 'AWS_CODECOMMIT', iconName: 'service-aws-code-deploy' },
  CUSTOM: { value: 'CUSTOM', iconName: 'build' }
}

const TriggerTypeIcons = {
  SCHEDULE: 'trigger-schedule'
}
export const getTriggerIcon = ({
  type,
  webhookSourceRepo
}: {
  type: string
  webhookSourceRepo: GetActionsListQueryParams['sourceRepo'] | undefined | string // string temporary until backend
}): IconName => {
  const webhookSourceRepoIconName =
    webhookSourceRepo && GitSourceProviders[webhookSourceRepo as GetActionsListQueryParams['sourceRepo']]?.iconName
  if (type === TriggerTypes.WEBHOOK && webhookSourceRepoIconName) {
    return webhookSourceRepoIconName as IconName
  } else if (type === TriggerTypes.SCHEDULE) {
    return TriggerTypeIcons.SCHEDULE as IconName
  }
  // placeholder for now
  return GitSourceProviders.GITHUB?.iconName as IconName
}

const triggerDrawerMap = (getString: (key: StringKeys) => string): AddDrawerMapInterface => ({
  drawerLabel: getString('pipeline-triggers.triggersLabel'),
  drawerSubLabel: getString('pipeline-triggers.triggersSubLabel'),
  showAllLabel: getString('pipeline-triggers.showAllTriggers'),
  searchPlaceholder: getString('pipeline-triggers.searchPlaceholder'),
  categories: [
    // {
    //   categoryLabel: getString('pipeline-triggers.onNewArtifactTitle'),
    //   categoryValue: 'OnArtifact',
    //   items: [
    //     {
    //       itemLabel: getString('pipeline-triggers.newArtifactLabel'),
    //       value: 'NewArtifact',
    //       iconName: 'trigger-artifact' as IconName
    //     },
    //     {
    //       itemLabel: getString('pipeline-triggers.newManifestLabel'),
    //       value: 'NewManifest',
    //       iconName: 'trigger-artifact' as IconName
    //     }
    //   ]
    // },
    {
      categoryLabel: getString('pipeline-triggers.scheduledLabel'),
      categoryValue: 'Scheduled',
      items: [
        {
          itemLabel: getString('pipeline-triggers.cronLabel'),
          value: 'Cron',
          iconName: TriggerTypeIcons.SCHEDULE as IconName
        }
      ]
    },
    {
      categoryLabel: getString('execution.triggerType.WEBHOOK'),
      categoryValue: 'Webhook',
      items: [
        {
          itemLabel: getString('common.repo_provider.githubLabel'),
          value: GitSourceProviders.GITHUB.value,
          iconName: GitSourceProviders.GITHUB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.gitlabLabel'),
          value: GitSourceProviders.GITLAB.value,
          iconName: GitSourceProviders.GITLAB.iconName
        },
        {
          itemLabel: getString('common.repo_provider.bitbucketLabel'),
          value: GitSourceProviders.BITBUCKET.value,
          iconName: GitSourceProviders.BITBUCKET.iconName
        },
        {
          itemLabel: getString('common.repo_provider.codecommit'),
          value: GitSourceProviders.AWS_CODECOMMIT.value,
          iconName: GitSourceProviders.AWS_CODECOMMIT.iconName
        },
        {
          itemLabel: getString('common.repo_provider.customLabel'),
          value: GitSourceProviders.CUSTOM.value,
          iconName: GitSourceProviders.CUSTOM.iconName
        }
      ]
    }
  ]
})

export const getSourceRepoOptions = (getString: (str: StringKeys) => string): { label: string; value: string }[] => [
  { label: getString('common.repo_provider.githubLabel'), value: GitSourceProviders.GITHUB.value },
  { label: getString('common.repo_provider.gitlabLabel'), value: GitSourceProviders.GITLAB.value },
  { label: getString('common.repo_provider.bitbucketLabel'), value: GitSourceProviders.BITBUCKET.value },
  { label: getString('common.repo_provider.codecommit'), value: GitSourceProviders.AWS_CODECOMMIT.value },
  { label: getString('common.repo_provider.customLabel'), value: GitSourceProviders.CUSTOM.value }
]

export const getCategoryItems = (getString: (key: StringKeys) => string): AddDrawerMapInterface =>
  triggerDrawerMap(getString)

export interface ItemInterface {
  itemLabel: string
  iconName: IconName
  value: string
  visible?: boolean
  disabled?: boolean
  categoryValue?: string
}

export interface TriggerDataInterface {
  triggerType: string
  sourceRepo?: string
  // all else optional
}
