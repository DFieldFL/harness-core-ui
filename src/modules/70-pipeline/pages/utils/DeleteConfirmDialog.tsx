import React, { useState } from 'react'
import { defaultTo, pick } from 'lodash-es'
import { useParams } from 'react-router-dom'
import { Intent, TextArea } from '@blueprintjs/core'
import { Text, useConfirmationDialog, useToaster } from '@wings-software/uicore'
import { EntityGitDetails, useDeleteInputSetForPipeline, useSoftDeletePipeline } from 'services/pipeline-ng'
import { useStrings } from 'framework/strings'
import { useDeleteTemplateVersionsOfIdentifier } from 'services/template-ng'
import type { ResponseBoolean } from 'services/cd-ng'
import type { PipelineDTO } from '../pipelines/views/PipelineListView'
import type { InputSetLocal } from '../inputSet-list/InputSetListView'

const useGetEntityText = (entityType: string): string => {
  const { getString } = useStrings()

  switch (entityType) {
    case 'pipeline':
      return getString('common.pipeline')
    case 'inputSet':
      return getString('inputSets.inputSetLabel')
    case 'overlayInputSet':
      return getString('inputSets.overlayInputSet')
    default:
      return ''
  }
}
interface DeleteConfirmDialogContentProps {
  gitDetails?: EntityGitDetails
  entityName: string
  commitMsg: string
  onCommitMsgChange: (commitMsg: string) => void
  entityType: string
}

export const DeleteConfirmDialogContent: React.FC<DeleteConfirmDialogContentProps> = ({
  gitDetails = {},
  entityName = '',
  commitMsg,
  onCommitMsgChange,
  entityType = ''
}): JSX.Element => {
  const { getString } = useStrings()
  return (
    <div className={'pipelineDeleteDialog'}>
      <Text margin={{ bottom: 'medium' }} title={entityName}>{`${getString(
        'common.git.confirmDelete'
      )} ${useGetEntityText(entityType)} ${entityName}?`}</Text>
      {gitDetails?.objectId && (
        <>
          <Text>{getString('common.git.commitMessage')}</Text>
          <TextArea
            value={commitMsg}
            onInput={(event: React.ChangeEvent<HTMLTextAreaElement>) => {
              onCommitMsgChange(event.target.value)
            }}
          />
        </>
      )}
    </div>
  )
}

interface DeleteMetaData {
  versions?: string[]
}

export default function useDeleteConfirmationDialog(
  entityData: PipelineDTO | InputSetLocal,
  entityType: string,
  refetchData?: () => void,
  pipelineIdentifier = '',
  setLoading?: (isLoading: boolean) => void
): { confirmDelete: (deleteMetaData?: DeleteMetaData) => void } {
  const { showSuccess, showError } = useToaster()
  const { getString } = useStrings()

  const { projectIdentifier, orgIdentifier, accountId } = useParams<{
    projectIdentifier: string
    orgIdentifier: string
    accountId: string
  }>()

  const [deleteCallMetaData, setDeleteCallMetaData] = useState<DeleteMetaData>()

  const getSuccessMessage = (): string => {
    switch (entityType) {
      case 'pipeline':
        return getString('pipeline-list.pipelineDeleted', { name: entityData.name })
      case 'inputSet':
      case 'overlayInputSet':
        return getString('inputSets.inputSetDeleted', { name: entityData.name })
      case 'template':
        return getString('common.template.deleteTemplate.templatesDeleted', { name: entityData.name })
      default:
        return ''
    }
  }

  const getErrorMessage = (): string => {
    switch (entityType) {
      case 'pipeline':
        return `pipeline.delete.pipeline.error`
      case 'inputSet':
      case 'overlayInputSet':
        return `pipeline.delete.inputset.error`
      case 'template':
        return `common.template.deleteTemplate.errorWhileDeleting`
      default:
        return ''
    }
  }

  const [commitMsg, setCommitMsg] = React.useState<string>(`${getString('delete')} ${entityData.name}`)

  const gitParams = entityData.gitDetails?.objectId
    ? {
        ...pick(entityData.gitDetails, ['branch', 'repoIdentifier', 'filePath', 'rootFolder']),
        commitMsg,
        lastObjectId: entityData.gitDetails?.objectId
      }
    : {}

  const { mutate: deletePipeline } = useSoftDeletePipeline({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier,
      ...gitParams
    }
  })

  const { mutate: deleteInputSet } = useDeleteInputSetForPipeline({
    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier, pipelineIdentifier, ...gitParams }
  })

  const { mutate: deleteTemplates } = useDeleteTemplateVersionsOfIdentifier({})

  const deleteEntity = async (): Promise<ResponseBoolean> => {
    const contentType = 'application/json'
    let deleted = null
    if (entityType === 'pipeline') {
      deleted = await deletePipeline(entityData.identifier || /* istanbul ignore next */ '', {
        headers: { 'content-type': contentType }
      })
    } else if (entityType === 'template') {
      deleted = await deleteTemplates(defaultTo(entityData?.identifier, ''), {
        queryParams: {
          accountIdentifier: accountId,
          orgIdentifier,
          projectIdentifier,
          ...gitParams
        },
        body: JSON.stringify({ templateVersionLabels: deleteCallMetaData?.versions }),
        headers: { 'content-type': contentType }
      })
    } else {
      deleted = await deleteInputSet(entityData.identifier || /* istanbul ignore next */ '', {
        headers: { 'content-type': contentType }
      })
    }
    return deleted
  }

  const { openDialog: openConfirmDeleteDialog } = useConfirmationDialog({
    contentText: (
      <DeleteConfirmDialogContent
        entityName={entityData?.name || ''}
        entityType={entityType}
        gitDetails={entityData.gitDetails}
        commitMsg={commitMsg}
        onCommitMsgChange={setCommitMsg}
      />
    ),
    titleText: `${getString('delete')} ${useGetEntityText(entityType)}`,
    confirmButtonText: getString('delete'),
    cancelButtonText: getString('cancel'),
    intent: Intent.DANGER,
    buttonIntent: Intent.DANGER,
    onCloseDialog: async (isConfirmed: boolean) => {
      /* istanbul ignore else */
      if (isConfirmed) {
        try {
          setLoading?.(true)
          const deleted = await deleteEntity()
          setLoading?.(false)
          /* istanbul ignore else */
          if (deleted?.status === 'SUCCESS') {
            showSuccess(getSuccessMessage())
          } else {
            throw getString('somethingWentWrong')
          }
          refetchData?.()
        } catch (err) {
          /* istanbul ignore next */
          showError(err?.data?.message || err?.message, undefined, getErrorMessage())
        }
      }
    }
  })

  const confirmDelete = (deleteMetaData?: DeleteMetaData): void => {
    setDeleteCallMetaData(deleteMetaData)
    openConfirmDeleteDialog()
  }
  return { confirmDelete }
}
