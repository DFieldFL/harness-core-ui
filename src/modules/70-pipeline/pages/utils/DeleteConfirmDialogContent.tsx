import React from 'react'
import { TextArea } from '@blueprintjs/core'
import { Text } from '@wings-software/uicore'
import type { EntityGitDetails } from 'services/pipeline-ng'
import { useStrings } from 'framework/strings'

interface DeleteConfirmDialogContentProps {
  gitDetails?: EntityGitDetails
  entityName: string
  commitMsg: string
  onCommitMsgChange: (commitMsg: string) => void
  entityType: string
}

export const DeleteConfirmDialogContent: React.FC<DeleteConfirmDialogContentProps> = ({
  gitDetails = {},
  entityName = '',
  commitMsg,
  onCommitMsgChange,
  entityType = ''
}): JSX.Element => {
  const { getString } = useStrings()

  return (
    <div className={'pipelineDeleteDialog'}>
      <Text margin={{ bottom: 'medium' }} title={entityName}>{`${getString(
        'pipeline-list.confirmDelete'
      )} ${entityType} ${entityName}?`}</Text>
      {gitDetails?.objectId && (
        <>
          <Text>{getString('common.git.commitMessage')}</Text>
          <TextArea
            value={commitMsg}
            onInput={(event: React.ChangeEvent<HTMLTextAreaElement>) => {
              onCommitMsgChange(event.target.value)
            }}
          />
        </>
      )}
    </div>
  )
}
