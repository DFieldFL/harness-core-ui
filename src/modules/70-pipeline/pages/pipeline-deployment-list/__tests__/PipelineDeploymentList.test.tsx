import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { accountPathProps, pipelinePathProps } from '@common/utils/routeUtils'
import routes from '@common/RouteDefinitions'
import { defaultAppStoreValues } from '@projects-orgs/pages/projects/__tests__/DefaultAppStoreData'

import PipelineDeploymentList from '../PipelineDeploymentList'
import data from './pipeline-list.json'

jest.mock('@common/components/YAMLBuilder/YamlBuilder', () => ({ children }: { children: JSX.Element }) => (
  <div>{children}</div>
))

jest.mock('services/cd-ng', () => ({
  useGetListOfExecutions: jest.fn(() => data),
  useHandleInterrupt: jest.fn(() => ({}))
}))

describe('Test Pipeline Deployment list', () => {
  beforeAll(() => {
    jest.spyOn(global.Date, 'now').mockReturnValue(1603645966706)
  })
  afterAll(() => {
    jest.spyOn(global.Date, 'now').mockReset()
  })

  // eslint-disable-next-line jest/no-disabled-tests
  test('should render deployment list', async () => {
    const { container, getByText } = render(
      <TestWrapper
        path={routes.toCDPipelineDeploymentList({ ...accountPathProps, ...pipelinePathProps })}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <PipelineDeploymentList onRunPipeline={jest.fn()} />
      </TestWrapper>
    )
    await waitFor(() => getByText('test-p1'))
    expect(container).toMatchSnapshot()
  })
})
