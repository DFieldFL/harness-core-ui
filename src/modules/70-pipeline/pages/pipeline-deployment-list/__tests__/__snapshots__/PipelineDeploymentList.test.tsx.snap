// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pipeline Deployment list should render deployment list 1`] = `
<div>
  <div
    class="StyledProps---font--2ipy StyledProps---main--1Wws pageBody main"
  >
    <div
      class="main"
    >
      <button
        class="bp3-button StyledProps---font--2ipy Button---button--4ZD5 StyledProps---main--1Wws StyledProps---intent-primary--2Xgi"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-cube StyledProps---main--1Wws"
          icon="cube"
        >
          <svg
            data-icon="cube"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              cube
            </desc>
            <path
              d="M14.194 3.54L8 7.41 1.806 3.54 7.504.283a1 1 0 01.992 0l5.698 3.255zm.75.71a1 1 0 01.056.33v6.84a1 1 0 01-.504.868L8.5 15.714V8.277l6.444-4.027zm-13.888 0L7.5 8.277v7.437l-5.996-3.426A1 1 0 011 11.42V4.58a1 1 0 01.056-.33z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          <span
            class="runText"
          >
            Run Pipeline
          </span>
        </span>
      </button>
    </div>
    <div
      class="main"
    >
      <div
        class="bp3-card bp3-elevation-0 Card---card--2BXa card"
      >
        <div
          class="icons"
          data-cd="true"
          data-ci="false"
        >
          <span
            class="bp3-icon StyledProps---main--1Wws"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 26 21"
              width="20"
            >
              <path
                clip-rule="evenodd"
                d="M15.201 2.9a5.325 5.325 0 00-2.498 3.256l-.34 1.273 2.656.712.34-1.273a2.615 2.615 0 013.2-1.847 2.617 2.617 0 011.847 3.2 2.614 2.614 0 01-3.199 1.847l-.03-.008-7.673-2.057a5.325 5.325 0 00-4.069.535 5.325 5.325 0 00-2.497 3.256 5.328 5.328 0 00.536 4.07 5.328 5.328 0 003.256 2.499 5.325 5.325 0 004.069-.535 5.325 5.325 0 002.498-3.256l.34-1.274-2.656-.712-.34 1.273a2.614 2.614 0 01-3.2 1.847 2.617 2.617 0 01-1.847-3.2 2.604 2.604 0 011.218-1.585 2.604 2.604 0 011.98-.262l7.704 2.066a5.325 5.325 0 004.069-.535 5.325 5.325 0 002.497-3.257 5.329 5.329 0 00-.536-4.07 5.329 5.329 0 00-3.256-2.498 5.325 5.325 0 00-4.069.534z"
                fill="url(#cd-main_svg__paint0_linear)"
                fill-rule="evenodd"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="cd-main_svg__paint0_linear"
                  x1="3.557"
                  x2="8.848"
                  y1="9.622"
                  y2="18.787"
                >
                  <stop
                    stop-color="#62F91F"
                  />
                  <stop
                    offset="1"
                    stop-color="#5EAA53"
                  />
                </lineargradient>
              </defs>
            </svg>
          </span>
        </div>
        <div>
          <div
            class="content"
          >
            <div>
              <a
                class="title"
                href="/account/testAcc/cd/pipelines/orgs/testOrg/projects/test/pipelines/testp1/executions/TyNGXVdmRWi5vrOJopUKOg/pipeline"
              >
                <span
                  class="pipelineName"
                >
                  test-p1
                </span>
                <span
                  class="executionId"
                >
                  (Execution ID: DeploymentIdPlaceHolder)
                </span>
              </a>
              <div
                class="cdData"
              >
                <span
                  class="sectionTitle"
                >
                  Deployment
                </span>
                <div
                  class="servicesDeployed"
                >
                  <span>
                    Services Deployed (0)
                  </span>
                  <div
                    class="servicesList"
                  />
                </div>
              </div>
            </div>
            <div
              class="main"
            >
              <div
                class="graphWrapper"
              >
                <div
                  class="graph"
                >
                  <div
                    class="stage"
                    data-status="notstarted"
                  >
                    <span
                      class="bp3-icon StyledProps---main--1Wws icon"
                    >
                      <svg
                        fill="none"
                        height="13"
                        viewBox="0 0 36 36"
                        width="13"
                      >
                        <path
                          d="M18 2.084a15.915 15.915 0 010 31.831 15.915 15.915 0 010-31.83"
                          stroke="currentColor"
                          stroke-dasharray="75,100"
                          stroke-linecap="round"
                          stroke-width="2.8"
                          style="animation: 1s linear infinite spinner-rotation; transform-origin: 50% 50%; transform-box: fill-box;"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="stepCounts"
              >
                <div
                  class="stepCount"
                  data-status="success"
                >
                  <span
                    class="bp3-icon bp3-icon-tick-circle StyledProps---main--1Wws"
                    icon="tick-circle"
                  >
                    <svg
                      data-icon="tick-circle"
                      height="10"
                      viewBox="0 0 16 16"
                      width="10"
                    >
                      <desc>
                        tick-circle
                      </desc>
                      <path
                        d="M8 16c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4-11c-.28 0-.53.11-.71.29L7 9.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42l3 3c.18.18.43.29.71.29s.53-.11.71-.29l5-5A1.003 1.003 0 0012 5z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  0
                   / 
                </div>
                <div
                  class="stepCount"
                  data-status="failed"
                >
                  <span
                    class="bp3-icon bp3-icon-error StyledProps---main--1Wws"
                    icon="error"
                  >
                    <svg
                      data-icon="error"
                      height="10"
                      viewBox="0 0 16 16"
                      width="10"
                    >
                      <desc>
                        error
                      </desc>
                      <path
                        d="M7.99-.01c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm1 13h-2v-2h2v2zm0-3h-2v-7h2v7z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                   
                  0
                </div>
              </div>
            </div>
            <div
              class="actions"
            >
              <span
                class="StyledProps---font--2ipy StyledProps---main--1Wws status failed StyledProps---inline--1f52 StyledProps---font-weight-bold--22XZ StyledProps---font-size-xsmall--1f9b"
              >
                FAILED
              </span>
              <div
                class="main"
              >
                <button
                  class="bp3-button bp3-minimal bp3-small StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-repeat StyledProps---main--1Wws"
                    icon="repeat"
                  >
                    <svg
                      data-icon="repeat"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        repeat
                      </desc>
                      <path
                        d="M10 5c0 .55.45 1 1 1h4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1s-1 .45-1 1v1.74A7.95 7.95 0 008 0C3.58 0 0 3.58 0 8c0 4.06 3.02 7.4 6.94 7.92.02 0 .04.01.06.01.33.04.66.07 1 .07 4.42 0 8-3.58 8-8 0-.55-.45-1-1-1s-1 .45-1 1c0 3.31-2.69 6-6 6-.71 0-1.37-.15-2-.38v.01C3.67 12.81 2 10.61 2 8c0-3.31 2.69-6 6-6 1.77 0 3.36.78 4.46 2H11c-.55 0-1 .45-1 1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
                <span
                  class="bp3-popover-wrapper"
                >
                  <span
                    class="bp3-popover-target"
                  >
                    <button
                      class="bp3-button bp3-minimal bp3-small StyledProps---font--2ipy Button---button--4ZD5 Button---iconOnly--_pvt StyledProps---main--1Wws more"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-more StyledProps---main--1Wws"
                        icon="more"
                      >
                        <svg
                          data-icon="more"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            more
                          </desc>
                          <path
                            d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="footer"
          >
            <div
              class="triggerInfo"
            >
              <span
                class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---inline--1f52"
                style="display: inline-flex; align-items: center;"
              >
                <span
                  class="bp3-icon bp3-icon-user StyledProps---main--1Wws StyledProps---padding-right-xsmall--3D04"
                  icon="user"
                >
                  <svg
                    data-icon="user"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      user
                    </desc>
                    <path
                      d="M7.99-.01A7.998 7.998 0 00.03 8.77c.01.09.03.18.04.28.02.15.04.31.07.47.02.11.05.22.08.34.03.13.06.26.1.38.04.12.08.25.12.37.04.11.08.21.12.32a6.583 6.583 0 00.3.65c.07.14.14.27.22.4.04.07.08.13.12.2l.27.42.1.13a7.973 7.973 0 003.83 2.82c.03.01.05.02.07.03.37.12.75.22 1.14.29l.2.03c.39.06.79.1 1.2.1s.81-.04 1.2-.1l.2-.03c.39-.07.77-.16 1.14-.29.03-.01.05-.02.07-.03a8.037 8.037 0 003.83-2.82c.03-.04.06-.08.09-.13.1-.14.19-.28.28-.42.04-.07.08-.13.12-.2.08-.13.15-.27.22-.41.04-.08.08-.17.12-.26.06-.13.11-.26.17-.39.04-.1.08-.21.12-.32.04-.12.08-.24.12-.37.04-.13.07-.25.1-.38.03-.11.06-.22.08-.34.03-.16.05-.31.07-.47.01-.09.03-.18.04-.28.02-.26.04-.51.04-.78-.03-4.41-3.61-7.99-8.03-7.99zm0 14.4c-1.98 0-3.75-.9-4.92-2.31.67-.36 1.49-.66 2.14-.95 1.16-.52 1.04-.84 1.08-1.27.01-.06.01-.11.01-.17-.41-.36-.74-.86-.96-1.44v-.01c0-.01-.01-.02-.01-.02-.05-.13-.09-.26-.12-.39-.28-.05-.44-.35-.5-.63-.06-.11-.18-.38-.15-.69.04-.41.2-.59.38-.67v-.06c0-.51.05-1.24.14-1.72.02-.13.05-.26.09-.39.17-.59.53-1.12 1.01-1.49.49-.38 1.19-.59 1.82-.59.62 0 1.32.2 1.82.59.48.37.84.9 1.01 1.49.04.13.07.26.09.4.09.48.14 1.21.14 1.72v.07c.18.08.33.26.37.66.03.31-.1.58-.16.69-.06.27-.21.57-.48.62-.03.13-.07.26-.12.38 0 .01-.01.04-.01.04-.21.57-.54 1.06-.94 1.42 0 .06.01.13.01.19.04.43-.12.75 1.05 1.27.65.29 1.47.6 2.14.95a6.415 6.415 0 01-4.93 2.31z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                Admin
              </span>
              <span
                class="triggerType"
              >
                Manually
              </span>
            </div>
            <div
              class="timers"
            >
              <span
                class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---inline--1f52"
                style="display: inline-flex; align-items: center;"
              >
                <span
                  class="bp3-icon bp3-icon-time StyledProps---main--1Wws StyledProps---padding-right-xsmall--3D04"
                  icon="time"
                >
                  <svg
                    data-icon="time"
                    height="12"
                    viewBox="0 0 16 16"
                    width="12"
                  >
                    <desc>
                      time
                    </desc>
                    <path
                      d="M8 0C3.58 0 0 3.58 0 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm1-6.41V4c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .28.11.53.29.71l2 2a1.003 1.003 0 001.42-1.42L9 7.59z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                Duration: 
                2s
              </span>
              <span
                class="StyledProps---font--2ipy StyledProps---main--1Wws StyledProps---inline--1f52"
                style="display: inline-flex; align-items: center;"
              >
                <span
                  class="bp3-icon bp3-icon-calendar StyledProps---main--1Wws StyledProps---padding-right-xsmall--3D04"
                  icon="calendar"
                >
                  <svg
                    data-icon="calendar"
                    height="12"
                    viewBox="0 0 16 16"
                    width="12"
                  >
                    <desc>
                      calendar
                    </desc>
                    <path
                      d="M11 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1s-1 .4-1 1v1c0 .5.4 1 1 1zm3-2h-1v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H6v1c0 1.1-.9 2-2 2s-2-.9-2-2V1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h13c.6 0 1-.4 1-1V2c0-.6-.5-1-1-1zM5 13H2v-3h3v3zm0-4H2V6h3v3zm4 4H6v-3h3v3zm0-4H6V6h3v3zm4 4h-3v-3h3v3zm0-4h-3V6h3v3zM4 3c.6 0 1-.5 1-1V1c0-.6-.4-1-1-1S3 .4 3 1v1c0 .5.4 1 1 1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <time
                  datetime="2020-10-25T17:05:54.243Z"
                  title="2020-10-25 17:05"
                >
                  7 minutes ago
                </time>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="pagination"
    >
      <div
        class="Layout---horizontal--3rmQ StyledProps---main--1Wws Pagination---container--3ype StyledProps---padding-top-medium--29t2 StyledProps---padding-bottom-medium--3R7v StyledProps---flex--2Jz8 StyledProps---flex-align-center-center--2AmX StyledProps---flex-distribution-space-between--1TyN"
      >
        <span
          class="StyledProps---font--2ipy StyledProps---main--1Wws Pagination---text--30fh StyledProps---inline--1f52"
        >
          Showing 
          <strong>
            1
          </strong>
           of 
          <strong>
            1
          </strong>
        </span>
      </div>
    </div>
  </div>
</div>
`;
